playbooks/secrets.yml

```diff
@@ -0,0 +1,85 @@
+---
+- hosts: all:!external
+  gather_facts: false
+  tasks:
+    - name: Ensure no secrets are still provided via group_vars and/or hostvars
+      assert:
+        that:
+          - repo_db_password is not defined
+          - sync_db_password is not defined
+          - reposearch_shared_secret is not defined
+        msg:
+          "The current playbook version expect secrets to be provided in vars/secrets.yml.
+          If you are upgrading from a previous playbook version, please read to docs/playbook-upgrade.md"
+        quiet: true
+
+- hosts: localhost
+  gather_facts: false
+  vars:
+    base_folder: "{{ playbook_dir }}/.."
+    secrets_file: "{{ base_folder }}/vars/secrets.yml"
+  tasks:
+    - name: Ensure empty secrets.yml is present
+      ansible.builtin.file:
+        path: "{{ secrets_file }}"
+        state: touch
+        modification_time: preserve
+        access_time: preserve
+
+    - name: Load secrets as localhost hostvars
+      include_vars:
+        file: ../vars/secrets.yml
+
+    - name: Ensure that all mandatory secrets are properly set
+      assert:
+        that:
+          - repo_db_password is defined and repo_db_password | length > 0
+          - sync_db_password is defined and sync_db_password | length > 0
+          - reposearch_shared_secret is defined and reposearch_shared_secret | length > 0
+        msg: "Mandatory secrets are missing from vars/secrets.yml.
+          If you are using encrypted files you need to add them manually before the play."
+        quiet: true
+
+    - name: Ensure activemq_password is set when required
+      assert:
+        that: activemq_password is defined and activemq_password | length > 0
+        msg: "activemq_password must have been already set at this point.
+          If you are using encrypted files you need to add manually the secret before the play."
+        quiet: true
+      when: groups.external_activemq | default(False)
```

Comment:
We should have that even if not using external activemq

Reply:
is currently not possible to configure credentials with our activemq role (see the [next task](https://github.com/Alfresco/alfresco-ansible-deployment/pull/309/files#diff-90d24c1b1d1fdebb80688e0327d87085edba291742a1e7775f0d673ecd77d00fR81)) so there is no reason to set credential if not using external activemq

Reply:
OK I created OPSEXP-1459 and linked it to that same epic